pkg/config/config_linux.go

```diff
@@ -24,3 +33,115 @@ func (c *RuntimeConfig) ValidatePinnsPath(executable string) error {
 
 	return err
 }
+
+// checkKernelRROMountSupport checks the kernel support for the Recursive Read-only (RRO) mounts.
+func checkKernelRROMountSupport() error {
+	kernelRROSupportOnce.Do(func() {
+		kernelRROSupportError = (func() error {
+			// Check the current kernel version for RRO mounts support...
+			err := validateKernelRROVersion()
+			if err != nil {
+				versionErr := err
+				// ... and if the kernel version does not match the minimum required version,
+				// then verify whether the kernel supports RRO mounts regardless, as often
+				// Linux distributions provide heavily patched kernel releases, and the
+				// current kernel might include backported support.
+				err = validateKernelRROMount()
+				if err != nil {
+					err = fmt.Errorf("%w: %w", versionErr, err)
+				}
+			}
+			return err
+		})()
+	})
+
+	return kernelRROSupportError
+}
+
+// validateKernelRROVersion checks whether the current kernel version matches the release 5.12 or newer,
+// which is the minimum required kernel version that supports Recursive Read-only (RRO) mounts.
+func validateKernelRROVersion() error {
+	const kernelVersion = 512
+
+	var (
+		uname        unix.Utsname
+		major, minor int
+	)
+
+	if err := unix.Uname(&uname); err != nil {
+		return fmt.Errorf("unable to retrieve kernel version: %w", err)
+	}
+
+	index := bytes.IndexByte(uname.Release[:], 0)
+	version := string(uname.Release[:index])
+
+	_, err := fmt.Sscanf(version, "%d.%d", &major, &minor)
+	if err != nil {
+		return fmt.Errorf("unable to parse kernel version %q: %w", version, err)
+	}
+
+	supportedMajor := major > kernelVersion/100
+	supportedMinor := major == kernelVersion/100 && minor >= kernelVersion%100
+
+	if !supportedMajor && !supportedMinor {
+		return fmt.Errorf("kernel version %q does not support recursive read-only mounts", version)
+	}
+
+	return nil
+}
+
+// validateKernelRROMount checks whether the current kernel can support Recursive Read-only mounts.
+// It uses a test mount of tmpfs against which an attempt will be made to set the required attributes.
+// If there is no failure in doing so, then the kernel has the required support.
+func validateKernelRROMount() error {
+	path, err := os.MkdirTemp("", "crio-rro-*")
+	if err != nil {
+		return fmt.Errorf("unable to create directory: %w", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(path); err != nil {
+			logrus.Errorf("Unable to remove directory %q: %v", path, err)
```

Comment:
nit: os.RemoveAll error is like "rmdir %s: %w" (meaning you don't have to include the directory name and the fact that directory removal failed, as it's already in the error).

Reply:
Ah, yes. The `PathError` type captures it. Will change.

Reply:
Done.